src/ORM/Rule/ExistsIn.php

```diff
@@ -129,4 +146,22 @@ protected function _fieldsAreNull($entity, $source)
         }
         return $nulls === count($this->_fields);
     }
+
+    /**
+     * Check whether there are nullable nulls in at least one part of the foreign key.
+     *
+     * @param \Cake\Datasource\EntityInterface $entity The entity to check.
+     * @param \Cake\ORM\Table $source The table to use schema from.
+     * @return bool
+     */
+    protected function _checkPartialSchemaNulls($entity, $source)
+    {
+        $schema = $source->schema();
+        foreach ($this->_fields as $field) {
+            if ($schema->isNullable($field) && $entity->get($field) === null) {
+                return true;
+            }
```

Comment:
While I think its good practice to have less magic and I think the condition alone, that *a given NULLABLE key IS NULL* is sufficient (speaking in logics) for passing of the rule, people may forget to setup things correctly (see A)) and imply that B) is sufficient to include the first rule of A):

### A) Correct, explicit and strict setup, within `T2Table`:
```php
$rules->add($rules->existsIn(['y'], 'T1')); // does not pass whenever Y IS NULL
$rules->add($rules->existsIn(['x', 'y'], 'T1')); // passes whenever at least X IS NULL
```

### B) Desired, new implicit setup, within `T2Table`:
```php
$rules->add($rules->existsIn(['x', 'y'], 'T1')); // will only pass if Y is NOT NULL, while X may be either NULL OR NOT NULL
```
I'd still advertise to setup rules like in **A)** in the docs - good?

At the moment I cannot foresee any negative consequences by that small implicit magic - can you?

I will make the changes to the PR you both suggested @markstory and @chinpei215 as soon as I can.